Source/Particles/MultiParticleContainer.cpp

```diff
@@ -347,6 +348,21 @@ MultiParticleContainer::ReadParameters ()
 
         ParmParse pp_lasers("lasers");
         pp_lasers.queryarr("names", lasers_names);
+        auto const nlasers = lasers_names.size();
+        // Get lasers to deposit on main grid
+        m_laser_deposit_on_main_grid.resize(nlasers, false);
+        std::vector<std::string> tmp;
+        pp_lasers.queryarr("deposit_on_main_grid", tmp);
```

Comment:
loop over `lasers_names` instead and query each laser for `deposit_on_main_grid`? (see above)